### PR TITLE
Has many validator should work with simple arrays/objects

### PR DESCRIPTION
**Pull request by offirgolan**

### CODE REVIEW

File: tests/integration/validations/model-relationships-test.js [hunk 1]
```diff
@@ -23,6 +23,14 @@ var Validations = buildValidations({
   lastName: validator(Validators.presence)
 });
 
+var BelongsToValidations = buildValidations({
```
Comment (stefanpenner) [inline tests/integration/validations/model-relationships-test.js]: `const` ?

File: app/validators/has-many.js [hunk 2]
```diff
@@ -47,7 +64,7 @@ export default Base.extend({
           return emberArray(models).getEach('validations');
         });
       } else {
-        return value.toArray().getEach('validations');
+        return emberArray(value).getEach('validations');
```
Comment (stefanpenner) [inline app/validators/has-many.js]: `value.map(entry => get(entry, 'validations'))`

**Comment by offirgolan** [comment]
@stefanpenner this one is ready for review as well.

File: tests/integration/validations/model-relationships-test.js [hunk 3]
```diff
@@ -87,13 +87,59 @@ test("belong to validation - with cycle", function(assert) {
 
 });
 
-test("has-many relationship is async", function(assert) {
-  this.register('validator:has-many', HasManyValidator);
+test("has-many relationship is sync", function(assert) {
+  this.register('validator:belongs-to', BelongsToValidator);
+
+  var friend = setupObject(this, Ember.Object.extend(Validations), {
```
Comment (stefanpenner) [inline tests/integration/validations/model-relationships-test.js]: let

File: tests/integration/validations/model-relationships-test.js [hunk 4]
```diff
@@ -87,13 +87,59 @@ test("belong to validation - with cycle", function(assert) {
...
+    firstName: 'John'
+  });
+
+  var user = setupObject(this, Ember.Object.extend(HasManyValidations), {
```
Comment (stefanpenner) [inline tests/integration/validations/model-relationships-test.js]: let

File: tests/integration/validations/model-relationships-test.js [hunk 5]
```diff
@@ -87,13 +87,59 @@ test("belong to validation - with cycle", function(assert) {
...
+  assert.equal(friends.get('isValid'), false);
+  assert.equal(friends.get('message'), 'lastName should be present');
+});
+
+test("has-many relationship is sync with proxy", function(assert) {
+  this.register('validator:belongs-to', BelongsToValidator);
 
-  var HasManyValidations = buildValidations({
-    friends: validator('has-many')
+  var friend = setupObject(this, Ember.Object.extend(Validations), {
```
Comment (stefanpenner) [inline tests/integration/validations/model-relationships-test.js]: let

File: tests/integration/validations/model-relationships-test.js [hunk 6]
```diff
@@ -87,13 +87,59 @@ test("belong to validation - with cycle", function(assert) {
...
+    firstName: 'John'
   });
 
+  var user = setupObject(this, Ember.Object.extend(HasManyValidations), {
```
Comment (stefanpenner) [inline tests/integration/validations/model-relationships-test.js]: let

**Comment by stefanpenner** [comment]
small nit-picks, but otherwise :+1:

**Pull request merged**